src/Security/TwoFactor/BackendProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\TwoFactor;
+
+use Contao\User;
+use Scheb\TwoFactorBundle\Security\TwoFactor\AuthenticationContextInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Provider\TwoFactorFormRendererInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Provider\TwoFactorProviderInterface;
+
+class BackendProvider implements TwoFactorProviderInterface
+{
+    /**
+     * @var Authenticator
+     */
+    private $authenticator;
+
+    /**
+     * @var BackendFormRenderer
+     */
+    private $renderer;
+
+    /**
+     * @var bool
+     */
+    private $enforce2fa;
+
+    /**
+     * @param Authenticator       $authenticator
+     * @param BackendFormRenderer $renderer
+     * @param bool                $enforce2fa
+     */
+    public function __construct(Authenticator $authenticator, BackendFormRenderer $renderer, bool $enforce2fa)
+    {
+        $this->authenticator = $authenticator;
+        $this->renderer = $renderer;
+        $this->enforce2fa = $enforce2fa;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function beginAuthentication(AuthenticationContextInterface $context): bool
```

Comment:
The `AuthenticationContextInterface` also provides the firewall name and token. I think we should validate for `UsernamePasswordToken` and `backend` firewall (or match the user object). We only want 2FA for the actual login route, never for any other means of authentication, should the firewall be configured differently.